packages/doenetml-worker/lib-doenetml-core/src/dast/ref_resolve.rs

```diff
@@ -461,6 +498,29 @@ impl Resolver {
     }
 }
 
+/// Get the name of `element` from its `name` attribute,
+/// where the `name` attribute must have exactly one text child to be considered valid.
+fn get_element_name(element: &FlatElement) -> Option<String> {
+    // XXX: we need to enforce restrictions on valid names. Is this the right place?
```

Comment:
I think we should allow all names when we get to the rust code. Any validation of names should happen in the `normalize_dast` phase

Reply:
OK